packages/template-retail-react-app/app/hooks/use-product-view-modal.js

```diff
@@ -5,30 +5,54 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
-import {useEffect, useState} from 'react'
+import {useEffect} from 'react'
 import {rebuildPathWithParams, removeQueryParamsFromPath} from '../utils/url'
 import {useHistory, useLocation} from 'react-router-dom'
-import {useCommerceAPI} from '../commerce-api/contexts'
-import {isError} from '../commerce-api/utils'
 import {useVariant} from './use-variant'
 import {useToast} from './use-toast'
 import {useIntl} from 'react-intl'
 import {API_ERROR_MESSAGE} from '../constants'
+import {useProduct} from 'commerce-sdk-react-preview'
 
 /**
- * This hooks is responsible for fetching a product detail based on the variation selection
+ * This hook is responsible for fetching a product detail based on the variation selection
  * and managing the variation params on the url when the modal is open/close
  * @param initialProduct - the initial product when the modal is first open
  * @returns object
  */
 export const useProductViewModal = (initialProduct) => {
     const location = useLocation()
-    const api = useCommerceAPI()
     const history = useHistory()
     const intl = useIntl()
-    const [product, setProduct] = useState(initialProduct)
-    const [isFetching, setIsFetching] = useState(false)
     const toast = useToast()
+    const urlParams = new URLSearchParams(location.search)
+    const pid = urlParams.get('pid')
+    // use commerce-sdk-react-preview useProduct to fetch new product detail whenever variant changes
+    const {data: product, isFetching} = useProduct(
+        {id: pid},
+        {
+            // Show initialTodos immediately, but won't refetch until another interaction event is encountered after 1000 ms
```

Comment:
This mentions `initialTodos` but I don't think that's accurate.

Also, is the "won't refetch until another interaction event is encountered after 1000 ms" portion universally true of all hooks library fetch events? If so, we don't want to repeat this comment every time we call a hook

Reply:
Good catch, I've made some change to not use initialData since it is persisted with cache where it will trigger unwanted behavior when users switch variants in the modal. Using placeHolderData is the ideal which is to populate the current basketItemProduct into the modal.